packages/back-end/src/models/SdkPayloadModel.ts

```diff
@@ -83,6 +85,14 @@ export async function updateSDKPayload({
     experiments: experimentsDefinitions,
   };
 
+  const projectDoc = project
+    ? await findProjectById(project, organization)
+    : null;
+  if (project && !projectDoc) {
+    logger.error(`Cannot update SDK payload: project ${project} not found`);
+    return;
+  }
```

Comment:
Is there a way we can catch this earlier?  If someone deletes a project that's used by 20 features, it will call `onFeatureUpdated` 20 times.  Each of those is going to call this function with the now-deleted project.  And that means 20 additional Mongo queries checking if the project exists and 20 error logs, even though it's expected behavior and not an error.

Plus, every feature update in a valid project will also run an additional Mongo query now.

Reply:
Thanks for the catch, updated.